xla/mlir/tools/mlir_interpreter/dialects/comparators.h

```diff
@@ -43,14 +44,21 @@ struct FloatCompare : CwiseAll {
     }
   }
 
+  // Template type for non-floating point and non-complex
   template <typename T>
   static bool isnan(T a) {
-    return std::isnan(a);
+    return false;
   }
+  // Handling complex types
   template <typename T>
   static bool isnan(std::complex<T> a) {
     return std::isnan(std::real(a)) || std::isnan(std::imag(a));
   }
+  // Template for isnan (Floating numbers)
+  template <typename T>
+  static bool isnan(std::is_floating_point<T> a) {
+    return std::isnan(a);
+  }
```

Comment:
I don't see how this should work. `std::is_floating_point<T>` is not a placeholder for any floating point type.

(https://gcc.godbolt.org/z/Ej86a8jKK)